File: twoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    var i,j,h = {}, results=[];
    for(i=0; i<nums.length; i++) {
        j = target - nums[i];
        if(h[''+j] !== undefined) {
            results.push(h[''+j], i+1);
            break;
        } else {
            h[''+nums[i]] = i+1;
        }
    }
    return results;
};

console.log(twoSum([3,2,4], 6));

File: two_sum.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
    h = {}
    nums.each_with_index do |e, i|
        puts e, i , h
        if h[target-e]
            return [h[target-e] + 1, i + 1]
        else
            h[e] = i
        end
    end
end

two_sum([3,2,4], 6)
